### PR TITLE
Separate all tests in searching folder and some others

### PR DESCRIPTION
**Pull request by arnavb**
Self explanatory. Basically like what I did in #68. Add some tests as well.

### CODE REVIEW

**Comment by arnavb** [comment]
@AnshulMalik I've implemented/separated more tests and this should be ready to merge once CI passes.

File: src/sorting/heap_sort.rs [hunk 1]
```diff
@@ -122,4 +122,32 @@ mod tests {
...
+
+    #[test]
+    fn unusual_element_at_end() {
+        let mut arr = vec![3, 4, 2, 1, 7, 33333];
+        heap_sort(&mut arr);
+        assert_eq!(&arr, &[1, 2, 3, 4, 7, 33333]);
+    }
+
+    #[test]
+    fn unusual_element_at_start() {
```
Comment (AnshulMalik) [inline src/sorting/heap_sort.rs]: Do we need these unusual tests?

Comment (arnavb) [reply src/sorting/heap_sort.rs]: @AnshulMalik I don't know if we _need_ them, but I've learned that when testing, unusual end cases should definitely be considered.

Comment (arnavb) [reply src/sorting/heap_sort.rs]: However, if you feel that these tests are unnecessary, I can remove them.

Comment (AnshulMalik) [reply src/sorting/heap_sort.rs]: Yep, feels like redundant to me. Better if we keep them concise :)

**Pull request merged**